File: server/app/schemas/player.py
from datetime import date, datetime

from pydantic import BaseModel


class PlayerOut(BaseModel):
    id: int
    external_id: str
    name: str
    nationality: str | None = None
    tour: str | None = None
    created_at: datetime

    model_config = {"from_attributes": True}


class PlayerStatOut(BaseModel):
    id: int
    player_id: int
    event_date: date
    scoring_average: float | None = None
    strokes_gained_total: float | None = None
    driving_accuracy: float | None = None
    greens_in_regulation: float | None = None
    putting_average: float | None = None
    created_at: datetime

    model_config = {"from_attributes": True}

File: client/src/types/player.ts
export interface Player {
  id: number;
  external_id: string;
  name: string;
  nationality: string | null;
  tour: string | null;
  created_at: string;
}

export interface PlayerStat {
  id: number;
  player_id: number;
  event_date: string;
  scoring_average: number | null;
  strokes_gained_total: number | null;
  driving_accuracy: number | null;
  greens_in_regulation: number | null;
  putting_average: number | null;
  created_at: string;
}
